dissect/target/plugins/os/unix/bsd/osx/network.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+
+import plistlib
+from functools import cache
+from typing import Iterator
+
+from dissect.target.helpers.record import MacInterfaceRecord
+from dissect.target.plugins.general.network import NetworkPlugin
+
+
+class MacNetworkPlugin(NetworkPlugin):
+    @cache
+    def _plistlease(self) -> None:
+        if (dhcp := self.target.fs.path("/private/var/db/dhcpclient/leases")).exists():
+            for lease in dhcp.iterdir():
+                if lease.is_file():
+                    return plistlib.load(lease.open())
+        return {}
+
+    @cache
```

Comment:
Nice, now can you also change this to how you should use the cache decorator on a class? Do a `self._whatever = cache(self._whatever)` in the `__init__`.